### PR TITLE
Add workaround for distributed service identities being invalid

### PR DESCRIPTION
**Pull request by rnicoll**
Distributed services currently have invalid certificate paths (they do not follow the name constraints of their signing certificate), so we can't register them. This adds a workaround until we can fix the underlying identity generation process.

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 1]
```diff
@@ -662,9 +663,18 @@ abstract class AbstractNode(open val configuration: NodeConfiguration,
                 .filterNotNull()
                 .toTypedArray()
         val service = PersistentIdentityService(setOf(info.legalIdentityAndCert), trustRoot = trustRoot, caCertificates = *caCertificates)
-        services.networkMapCache.partyNodes.forEach {
-            service.verifyAndRegisterIdentity(it.legalIdentityAndCert)
-            it.advertisedServices.forEach { service.verifyAndRegisterIdentity(it.identity) }
+        services.networkMapCache.partyNodes.forEach { node ->
+            service.verifyAndRegisterIdentity(node.legalIdentityAndCert)
+            node.advertisedServices.forEach { advertisedService ->
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Advertised services are going so this is probably the wrong thing to enumerate.

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 2]
```diff
@@ -662,9 +663,18 @@ abstract class AbstractNode(open val configuration: NodeConfiguration,
...
+                // of their signing certificate), so we can't register them.
+                // TODO: Fix distributed service identity generation then remove this try-catch
+                try {
+                    service.verifyAndRegisterIdentity(advertisedService.identity)
+                } catch(ex: CertPathValidatorException) {
+                    log.warn("Could not register service identity ${advertisedService.identity.name} with network map cache", ex)
+                }
+            }
         }
         services.networkMapCache.changed.subscribe { mapChange ->
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: We also have to fix up the equivalent code in MockNode for tests

Comment (joeldudleyr3) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Thanks

Comment (mnesbit) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Also, we need a unit test

Comment (mnesbit) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: I think the legalIdentitiesAndCerts field will contain all non-primary identities. However, with Kasia off ill you will have to test this.

Comment (mnesbit) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Can we kill this PR as a dupe of https://github.com/corda/corda/pull/1442

**Comment by rnicoll** [comment]
It's meant to patch up #1442, but as I think we're fixing things elsewhere now, closing

**Pull request closed without merging**